### PR TITLE
[MA : Completed Functionality of TODO App]

### PR DESCRIPTION
**Pull request by mihirarora28082000**

### CODE REVIEW

File: lib/presentation/widgets/task_card.dart [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+    super.key,
+    required this.task,
+  });
+
+  @override
+  State<TaskCard> createState() => _TaskCardState();
+}
+
+class _TaskCardState extends State<TaskCard> {
+  Color color(int priority) {
```
Comment (Ayush-gupta-josh) [inline lib/presentation/widgets/task_card.dart]: Improve function naming convention

File: lib/presentation/widgets/task_card.dart [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+    if (priority == 1) {
```
Comment (Ayush-gupta-josh) [inline lib/presentation/widgets/task_card.dart]: what does 1 represent? Using descriptive names for contact value

File: lib/presentation/widgets/task_card.dart [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+
+  @override
+  Widget build(BuildContext context) {
+    return !kIsWeb
+        ? Dismissible(
+            key: Key(widget.task.id.toString()),
+            direction: DismissDirection.startToEnd,
+            background: !widget.task.isCompleted
+                ? Container(
+                    color: Colors.green,
```
Comment (Ayush-gupta-josh) [inline lib/presentation/widgets/task_card.dart]: Note: Ideally Frontend uses a theme structure these colors should be passed through theme. C

File: lib/presentation/widgets/task_card.dart [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+                    color: Colors.red,
+                    child: const Icon(
+                      Icons.today_outlined,
+                      color: Colors.white,
+                      size: 30,
+                    )),
+            onDismissed: (direction) {
+              context.read<TodoBloc>().add(ChangeTaskState(task: widget.task));
+              !widget.task.isCompleted
+                  ? snackBar(context, 'Task Marked as Completed !', true)
```
Comment (Ayush-gupta-josh) [inline lib/presentation/widgets/task_card.dart]: Static text should be stored in constant

File: lib/presentation/widgets/task_card.dart [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+                                Icons.edit,
+                                color: Theme.of(context).primaryColor,
+                                size: ScalingQuery(context).fontSize(3),
+                              ),
+                              onPressed: () {
+                                showModal(context, setState,
+                                    descriptionText: widget.task.description,
+                                    titleText: widget.task.name,
+                                    sliderValueDefault:
+                                        4 - widget.task.priority.toDouble(),
```
Comment (Ayush-gupta-josh) [inline lib/presentation/widgets/task_card.dart]: Try to make code readable

File: lib/presentation/widgets/snackbar.dart [hunk 6]
```diff
@@ -0,0 +1,11 @@
+import 'package:flutter/material.dart';
```
Comment (Ayush-gupta-josh) [inline lib/presentation/widgets/snackbar.dart]: Move this Util

File: lib/presentation/widgets/modal_bottom.dart [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+                                  isCompleted: false,
+                                )));
+                            Navigator.of(context).pop();
+                          },
+                    child: titleText == '' ? const Text('SUBMIT') : const Text('UPDATE')),
+              ],
+            ),
+          ),
+        );
+      }));
```
Comment (Ayush-gupta-josh) [inline lib/presentation/widgets/modal_bottom.dart:126]: Add limiting rule trailing comas

File: lib/presentation/screen/todo_screen.dart [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+        title: const Text('Todo App'),
+        actions: [
+          ElevatedButton(
+            onPressed: () {
+              showMyDialog(context,
+                      "Are you sure you want to delete all Outgoing and Completed tasks ?")
+                  .then((value) {
+                if (value == true) {
+                  context.read<TodoBloc>().add(DeleteAllTasks());
+                }
```
Comment (Ayush-gupta-josh) [inline lib/presentation/screen/todo_screen.dart]: break down this complete function into smaller functions

File: lib/bloc/todo_bloc.dart [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+
+  @override
+  TodoState? fromJson(Map<String, dynamic> json) {
+    return TodoState.fromMap(json);
+  }
+
+  @override
+  Map<String, dynamic>? toJson(TodoState state) {
+    return state.toMap();
+  }
```
Comment (Ayush-gupta-josh) [inline lib/bloc/todo_bloc.dart]: Why these are part of bloc files

**Review by Ayush-gupta-josh — CHANGES_REQUESTED** [review]

File: lib/bloc/todo_bloc.dart (hunk 9 above)
Comment (mihirarora28082000) [reply lib/bloc/todo_bloc.dart]: I am actually using HydratedBloc State here. These are the functions which it expects us to override.

File: lib/presentation/widgets/snackbar.dart (hunk 6 above)
Comment (mihirarora28082000) [reply lib/presentation/widgets/snackbar.dart]: Done

File: lib/presentation/widgets/task_card.dart (hunk 2 above)
Comment (mihirarora28082000) [reply lib/presentation/widgets/task_card.dart]: done

File: lib/presentation/widgets/task_card.dart (hunk 1 above)
Comment (mihirarora28082000) [reply lib/presentation/widgets/task_card.dart]: done

File: lib/presentation/widgets/task_card.dart (hunk 4 above)
Comment (mihirarora28082000) [reply lib/presentation/widgets/task_card.dart]: done

File: lib/presentation/widgets/task_card.dart (hunk 5 above)
Comment (mihirarora28082000) [reply lib/presentation/widgets/task_card.dart]: done

File: lib/presentation/screen/todo_screen.dart (hunk 8 above)
Comment (mihirarora28082000) [reply lib/presentation/screen/todo_screen.dart]: done

File: lib/presentation/widgets/modal_bottom.dart (hunk 7 above)
Comment (mihirarora28082000) [reply lib/presentation/widgets/modal_bottom.dart:126]: fixed

**Pull request merged**